### ISSUE TITLE
Broken CI

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [bids-standard/legacy-validator#1358]
it looks like "**Build & publish containers / Push Docker image to Docker Hub (pull_request)**" fails consistently.

> **Error:** Username and password required

**Comment by rwblair** [comment]
Hopefully #1357 fixes this. credentials weren't available to PRs from other remotes. Changed it so we only publish on release instead of on any PR against master.

**Comment by DimitriPapadopoulos** [comment]
Yes, the above issue seems to have been fixed, thank you.

However, "**ci/circleci: test**" fails too, and the error seems totally unrelated to the PR:
https://circleci.com/gh/bids-standard/bids-validator/9753?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by DimitriPapadopoulos** [comment]
The reported errors don't make any sense at all, they point to `bids-validator/cli.js:81:15`:
```
 FAIL  bids-validator/tests/cli.spec.js
  ● Console

    console.log
      undefined does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/valid_dataset does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/valid_dataset does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/empty_files does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/empty_files does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/valid_dataset does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/valid_dataset does not exist 

      at bids-validator/cli.js:81:15

    console.log
      /root/project/bids-validator/tests/data/valid_dataset does not exist

      at bids-validator/cli.js:81:15
```
I'm rebasing against `master` in the hope the issue might go away.

**Comment by DimitriPapadopoulos** [comment]
Rebasing doesn't help, see #1355.

**Comment by rwblair** [comment]
I haven't been able to figure this one out. Rerunning the tests via SSH into circle the tests pass. Also haven't been able to recreate locally.

**Comment by sappelhoff** [comment]
on an up-to-date `master`, after doing `npm ci` and `npm test`, I get the following:

```
> test
> node bids-validator/bin/test-submodule-exists && jest

 PASS  bids-validator/tests/tsv.spec.js (6.096 s)
 PASS  bids-validator/tests/nii.spec.js (6.082 s)
 PASS  bids-validator/tests/json.spec.js (6.192 s)
 PASS  bids-validator/tests/cli.spec.js (6.868 s)
  ● Console

    console.log
      undefined does not exist 

      at bids-validator/cli.js:81:15

    console.log
      bids-validator@1.8.5-dev.0

      at Object.BIDS (bids-validator/validators/bids/start.js:24:13)

    console.log
      bids-specification@disable

      at Object.BIDS (bids-validator/validators/bids/start.js:26:13)

    console.log
      bids-validator@1.8.5-dev.0

      at Object.BIDS (bids-validator/validators/bids/start.js:24:13)

    console.log
      bids-specification@disable

      at Object.BIDS (bids-validator/validators/bids/start.js:26:13)

    console.log
      bids-validator@1.8.5-dev.0

      at Object.BIDS (bids-validator/validators/bids/start.js:24:13)

    console.log
      bids-specification@disable

      at Object.BIDS (bids-validator/validators/bids/start.js:26:13)

    console.log
      bids-validator@1.8.5-dev.0

      at Object.BIDS (bids-validator/validators/bids/start.js:24:13)

    console.log
      bids-specification@disable

      at Object.BIDS (bids-validator/validators/bids/start.js:26:13)

 PASS  bids-validator/utils/files/__tests__/remoteFiles.spec.js
 PASS  bids-validator/tests/session.spec.js
 PASS  bids-validator/tests/bval.spec.js
 PASS  bids-validator/validators/tsv/__tests__/validateTsvColumns.spec.js
 PASS  bids-validator/tests/utils/files.spec.js
 PASS  bids-validator/tests/type.spec.js
 PASS  bids-validator/validators/__tests__/headerFields.spec.js
 PASS  bids-validator/tests/bvec.spec.js
 PASS  bids-validator/validators/bids/__tests__/checkDatasetDescription.spec.js
 PASS  bids-validator/tests/utils/config.spec.js
 PASS  bids-validator/validators/tsv/__tests__/checkPhenotype.spec.js
 PASS  bids-validator/utils/__tests__/filenamesOnly.spec.js
 PASS  bids-validator/utils/__tests__/type.spec.js
 PASS  bids-validator/tests/utils/files-web.spec.js
 PASS  bids-validator/utils/files/__tests__/readDir.spec.js
 PASS  bids-validator/validators/__tests__/checkAnyDataPresent.spec.js
 PASS  bids-validator/utils/__tests__/gitTreeMode.spec.js
 PASS  bids-validator/utils/files/__tests__/readDir-examples.spec.js
 PASS  bids-validator/validators/bids/__tests__/subjects.spec.js
 PASS  bids-validator/tests/headerField.spec.js
 PASS  bids-validator/utils/summary/__tests__/collectModalities.spec.js
 PASS  bids-validator/validators/nifti/__tests__/duplicateFiles.spec.js
 PASS  bids-validator/tests/consoleFormat.spec.js
 PASS  bids-validator/utils/summary/__tests__/collectPetFields.spec.js
 PASS  bids-validator/tests/utils/issues.spec.js
 PASS  bids-validator/utils/__tests__/collectSubjectMetadata.spec.js
 PASS  bids-validator/utils/__tests__/unit.spec.js
 PASS  bids-validator/tests/potentialLocations.spec.js
 PASS  bids-validator/utils/summary/__tests__/collectDatatype.spec.js
 PASS  bids-validator/utils/__tests__/bids_files.spec.js
 PASS  bids-validator/utils/summary/__tests__/collectSubjectMetadata.spec.js
 PASS  bids-validator/tests/events.spec.js (11.018 s)
 PASS  bids-validator/tests/bids.spec.js (17.574 s)
  ● Console

    console.log
      bids-validator@1.8.5-dev.0

      at Object.BIDS (bids-validator/validators/bids/start.js:24:13)

    console.log
      bids-specification@undefined

      at Object.BIDS (bids-validator/validators/bids/start.js:26:13)

 PASS  bids-validator/tests/bids-web.spec.js (18.085 s)
  ● Console

    console.log
      bids-validator@1.8.5-dev.0

      at Object.BIDS (bids-validator/validators/bids/start.js:24:13)

    console.log
      bids-specification@undefined

      at Object.BIDS (bids-validator/validators/bids/start.js:26:13)


Test Suites: 38 passed, 38 total
Tests:       682 passed, 682 total
Snapshots:   0 total
Time:        19.831 s
Ran all test suites.

```

**Comment by sappelhoff** [comment]
Interestingly, when running the CLI test **explicitly** instead of a blank `npm test`, it does fail Oo

```
,,,,,,,,,,,,,,,,,,,,/bids/bids-validator$ npm test bids-validator/tests/cli.spec.js 

> test
> node bids-validator/bin/test-submodule-exists && jest "bids-validator/tests/cli.spec.js"

 FAIL  bids-validator/tests/cli.spec.js
  CLI
    ✓ should display usage hints when no arguments / options are provided (43 ms)
    ✓ should accept a directory as the first argument without error (86 ms)
    ✓ without errors should exit with code 0 (24 ms)
    ✓ with errors should not exit with code 0 (4 ms)
    ✓ with errors should not exit with code 0 with --json argument (3 ms)
    ✓ should print valid json when the --json argument is provided (44 ms)
    ✓ should print with colors by default (31 ms)
    ✕ should print without colors when NO_COLOR env set (24 ms)

  ● CLI › should print without colors when NO_COLOR env set

    expect(received).not.toMatch(expected)

    Expected pattern: not /[[[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/
"

      94 |     process.env.NO_COLOR = 'any value'
      95 |     await cli(`${data_without_errors}`)
    > 96 |     expect(mockStdout.mock.calls[0][0]).not.toMatch(colorRegEx)
         |                                             ^
      97 |     delete process.env.NO_COLOR
      98 |   })
      99 | })

      at Object.<anonymous> (bids-validator/tests/cli.spec.js:96:45)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 7 passed, 8 total
Snapshots:   0 total
Time:        1.732 s, estimated 5 s
Ran all test suites matching /bids-validator\/tests\/cli.spec.js/i.
```

**Comment by bendhouseart** [comment]
EDIT: I did reproduce the error's circle ci and @sappelhoff encountered locally as well on osx.

I was hoping that something very different or glaring would pop up in the environment variables when running `npm test cli.spec.test.js` by itself versus running all tests at once with `npm test`, but so far that hasn't been the case for me. I captured the state of `process.env` and output it to a text file for each run case but the diff tells the real story, see below:

```bash
diff environment_variables_when_only_running_cli.spec.jst_test.txt environment_variables_when_running_all_tests.txt
38c38,39
<   VSCODE_INSPECTOR_OPTIONS: "{\"inspectorIpc\":\"/var/folders/zx/_0990xtn4y35ds688_jk_1wdzn8154/T/node-cdp.53253-6.sock\",\"deferredMode\":false,\"waitForDebugger\":\"\",\"execPath\":\"/usr/local/bin/node\",\"onlyEntrypoint\":false,\"autoAttachMode\":\"always\",\"openerId\":\"071bea744f5949f63dcf1d3b\"}",
---
>   _: "/Users/galassiae/Projects/bids-validator/node_modules/.bin/jest",
>   VSCODE_INSPECTOR_OPTIONS: "{\"inspectorIpc\":\"/var/folders/zx/_0990xtn4y35ds688_jk_1wdzn8154/T/node-cdp.53253-2.sock\",\"deferredMode\":false,\"waitForDebugger\":\"\",\"execPath\":\"/usr/local/bin/node\",\"onlyEntrypoint\":false,\"autoAttachMode\":\"always\",\"openerId\":\"7a03f18de53e1b0b37aaef6d\"}",
44c45
<   npm_lifecycle_event: "test",
---
>   npm_lifecycle_event: "jest-w",
63c64
<   npm_lifecycle_script: "node bids-validator/bin/test-submodule-exists && jest",
---
>   npm_lifecycle_script: "jest --watch",
78d78
<   _: "/Users/galassiae/Projects/bids-validator/node_modules/.bin/jest",
```

**Comment by sappelhoff** [comment]
fixed in #1386
